### PR TITLE
Upgrade to lightning 2.0

### LINKED ISSUE #1959: Test against lightning 2.0 release candidate

### PR DESCRIPTION
**Pull request by adamgayoso**
Fixes #1959 

Fixes autotune docs as well and makes hyperopt default searcher

### CODE REVIEW

**Comment by adamgayoso** [comment]
@martinkim0 this is mostly ready to go, but we should wait for the accelerator/devices changes.

File: tests/models/test_models.py [hunk 1]
```diff
@@ -192,7 +192,7 @@ def test_scvi(save_path):
         size_factor_key="size_factor",
     )
     model = SCVI(adata, n_latent=n_latent)
-    model.train(1, check_val_every_n_epoch=1, train_size=0.5)
+    model.train(12, check_val_every_n_epoch=1, train_size=0.5)
```
Comment (adamgayoso) [inline tests/models/test_models.py:195]:
```suggestion
    model.train(1, check_val_every_n_epoch=1, train_size=0.5)
```

**Comment by adamgayoso** [comment]
@martinkim0 the tuner test is failing now because validation loss is not being tracked, which seems to make sense to me as it's not tracked by default in SCVI. Do you know how it was being tracked before?

**Comment by adamgayoso** [comment]
ah I think it's an issue with the ray tune callback not being compatible with lightning 2.0

**Comment by adamgayoso** [comment]
https://github.com/ray-project/ray/issues/33426
